src/v/cluster/data_migration_backend.cc

```diff
@@ -1259,17 +1261,25 @@ ss::future<> backend::reconcile_migration(
                   enumerated_nts,
                   [this, &metadata, &mrstate](const auto& idx_nt) {
                       auto& [idx, nt] = idx_nt;
-                      auto& tstate = mrstate.outstanding_topics[nt];
-                      tstate.idx_in_migration = idx;
-                      _topic_migration_map.emplace(nt, metadata.id);
-                      return reconcile_topic(
-                        nt, tstate, metadata.id, mrstate.scope, true);
+                      return reconcile_topic(metadata.id, idx, nt, mrstate);
```

Comment:
> fix tstate lifetime

in the first commit, i have a guess about what the memory lifetime issue is that is being fixed, but i'm not certain. please state in the commit message what the issue is and how it's being fixed.